iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractTckSuiteExhaustivityTest.java

```diff
@@ -0,0 +1,84 @@
+
+public abstract class AbstractTckSuiteExhaustivityTest {
+
+    @Test
+    void assertAllTckTestsAreInstanced() {
+        ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+        String tckPackageName = "com.powsybl.iidm.network.tck";
+        List<String> tckTestClasses = getPackageClasses(classLoader, tckPackageName)
+                .filter(clazz -> containsTest(classLoader, clazz))
+                .filter(clazz -> !clazz.contains("$"))
+                .collect(toList());
+        assertNotEquals(0, tckTestClasses.size(), "Tck package is not loaded in this build");
+
+        String implementationPackageName = this.getClass().getPackageName();
+        Stream<String> implementationTestClasses = getPackageClasses(classLoader, implementationPackageName);
+
+        List<String> tckTestClassInheritors = implementationTestClasses
+                .map(res -> getAncestorFromPackage(classLoader, tckPackageName, res))
+                .filter(Optional::isPresent).map(Optional::get)
+                .collect(toList());
+
+        List<String> unimplementedTckTestClasses = tckTestClasses.stream()
+                .filter(clazz -> !tckTestClassInheritors.contains(clazz))
+                .collect(toList());
+
+        assertEquals(List.of(), unimplementedTckTestClasses,
+                "Some TCK test classes are not extended, so their test won't run\n");
+    }
+
+    private boolean containsTest(ClassLoader classLoader, String className) {
+        try {
+            return Arrays.stream(classLoader.loadClass(className)
+                    .getDeclaredMethods())
+                    .map(method -> method.isAnnotationPresent(Test.class)
+                            || method.isAnnotationPresent(TestFactory.class)
+                            || method.isAnnotationPresent(ParameterizedTest.class))
+                    .reduce(false, (a, b) -> a || b);
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Optional<String> getAncestorFromPackage(ClassLoader classLoader, String packageName, String className) {
+        try {
+            Class<?> clazz = classLoader.loadClass(className);
+            while (clazz.getSuperclass() != Object.class) {
+                clazz = clazz.getSuperclass();
+                if (clazz.getPackageName().startsWith(packageName)) {
+                    return Optional.of(clazz.getCanonicalName());
+                }
+            }
+            return Optional.empty();
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Stream<String> getPackageClasses(ClassLoader classLoader, String packageName) {
+        String packagePath = packageName.replace('.', '/');
+        InputStream tckTestResources = classLoader.getResourceAsStream(packagePath);
+        return new BufferedReader(new InputStreamReader(tckTestResources)).lines()
```

Comment:
To avoid a sonar issue:
```suggestion
        return new BufferedReader(new InputStreamReader(Objects.requireNonNull(tckTestResources))).lines()
```